runtime/src/cost_tracker.rs

```diff
@@ -174,22 +165,42 @@ impl CostTracker {
         (costliest_account, costliest_account_cost)
     }
 
-    fn would_fit(
+    fn would_fit(&self, tx_cost: &TransactionCost) -> Result<(), CostTrackerError> {
+        let mut writable_account = vec![];
+        writable_account.extend(&tx_cost.writable_accounts);
+        let mut cost = tx_cost.sum();
+        let mut account_data_size = tx_cost.account_data_size;
+        let mut vote_cost = if tx_cost.is_simple_vote { cost } else { 0 };
+
+        for tx_cost in self.pending_transactions.values() {
+            writable_account.extend(&tx_cost.writable_accounts);
```

Comment:
Having each `would_fit()` iterate over `pending_transactions` from all threads seems too expensive to me.

Also, this makes a big `writable_account` list and then checks if the sum cost of all `pending_transactions` fits into each of these account costs. That should be too restrictive.

Reply:
Mmm, maybe can cache the sum when adding to `pending_transaction`. 

The main reason for keeping individual transaction_cost in `pending_transaction` is to make it easier to apply `actual cu` instead of `estimated cu` for transactions.

`writable_account` may include duplicated account pubkey in this case, resulting additional check of same account. But not change the rules of how account limits work.

Reply:
Isn't `cost` the sum over all pending transaction costs and writable_accounts the union over writable accounts? So it would check each account in the union against the sum of all costs? If accounts are disjoint but the sun of the cost exceeds the account limit, it'd reject the transaction incorrectly.

Maybe we could apply the real cost by subtracting the estimate and adding the real cu?

Reply:
commit 521733f50783d556379ae2009366fe92295c4576 to remove pending_transactions, still leave plumbing works for `actual-units` in place